U/wmwv/plot valid pixels

wmwv opened the pull request:
Adds `valid_mask` option to `skyproj.draw_hspmap` to draw the valid pixel region of a HealSparseMap.

If passed a full rec_array, assumes valid_mask=True and emits a `UserWarning` that it is doing so and how to call with a component of the rec_array.

Add tests for new behavior, including verifying that the warning is emitted.

wmwv commented:
The warning intentionally raised now appears at the end of the pytest summary.  I don't understand how to prevent it from doing so.  I tried decorating the test function with
```
@pytest.mark.filterwarnings("ignore:draw_hspmap")
```
but that didn't work.

erykoff commented:
It will be a bit before I get to this, but are you sure that the warning isn't being triggered elsewhere?

wmwv commented:
Thank you!  You were right.  I have fixed the error and removed the unneeded pytest.mark decorator.  This was my first time using pytest to test that intended warnings were generated.  I understand a little better now how this is supposed to work.

wmwv commented:
Hmmm... now I'm wondering if the plotting range behavior is exactly right for valid_mask=True.  The valid_pixels are based on the NSIDE_COVERAGE, and so the plot defaults to the size of one NSIDE_COVERAGE cell.  That's different than the behavior if we plot a value from the HealSparseMap.  I don't quite understand why.

erykoff commented:
Okay, I will take a look at this.

wmwv commented:
Okay.  I standardized the three tests that should all produce the same "healpix_valid_pixels.png" to point to the same file (and removed the no-long used files).

erykoff reviewed (approved):
Looks good!
